test/ClangImporter/objc_bridging.swift

```diff
@@ -57,15 +59,44 @@ func mutablePointerToObjC(_ path: String) throws -> NSString {
 
 func objcStructs(_ s: StructOfNSStrings, sb: StructOfBlocks) {
   // Struct fields must not be bridged.
-  _ = s.nsstr! as Bool // expected-error {{cannot convert value of type 'Unmanaged<NSString>' to type 'Bool' in coercion}}
-
+  _ = s.nsstr! as Bool // expected-error {{cannot convert value of type 'NSString' to type 'Bool' in coercion}}
+  
   // FIXME: Blocks should also be Unmanaged.
-  _ = sb.block as Bool // expected-error {{cannot convert value of type '@convention(block) () -> Void' to type 'Bool' in coercion}}
+  _ = sb.block as Bool // expected-error {{cannot convert value of type '() -> Void' to type 'Bool' in coercion}}
   sb.block() // okay
+}
+
+// CHECK-IDE-TEST: struct StrongsInAStruct {
+// CHECK-IDE-TEST:   init(myobj: MYObject)
+// CHECK-IDE-TEST:   var myobj: MYObject
+// CHECK-IDE-TEST: }
+
+// CHECK-IDE-TEST: struct WeaksInAStruct {
+// CHECK-IDE-TEST:   init()
+// CHECK-IDE-TEST:   init(myobj: MYObject?)
+// CHECK-IDE-TEST:   weak var myobj: @sil_weak MYObject?
+// CHECK-IDE-TEST: }
+
+// CHECK-IDE-TEST: struct WeakAndNonnull {
+// CHECK-IDE-TEST: }
```

Comment:
I don't know how easy is to do, but besides the warning, this should not import at all. The layout that Swift has access to would be incorrect if we import it partially.

Reply:
Ah gotcha, shouldn't be too difficult. We exclude the whole struct already so could find a way to bail there.